### PR TITLE
AST-285: Add with-categories option

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: src/Command/ArgumentChecker.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (juliensnz) [inline src/Command/ArgumentChecker.php]: are you sure about that?

File: src/Command/ArgumentChecker.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Command;
+
+class ArgumentChecker
+{
+    /**
+     * This function checks if a value is allowed, and throw an exception if not.
```
Comment (SamirBoulil) [inline src/Command/ArgumentChecker.php]:
```suggestion
     * This function checks if a value is allowed, and throws an exception if not.
```

File: src/Command/ArgumentChecker.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     *
+     * @param string $value
+     * @param string $argumentName
+     * @param array $allowedValues
+     */
+    public static function check(string $value, string $argumentName, array $allowedValues)
```
Comment (SamirBoulil) [inline src/Command/ArgumentChecker.php]: assertOptionIsAllowed || assertAllowedOption ?

File: src/Command/CreateFamilyCommand.php [hunk 4]
```diff
@@ -57,22 +62,27 @@ protected function configure()
...
         $referenceType = $input->getOption('reference-type');
-        if (!in_array($referenceType, [self::LOCALIZABLE, self::NON_LOCALIZABLE, self::BOTH])) {
-            throw new \InvalidArgumentException(sprintf(
-                'Argument "reference-type" should be "%s", "%s" or "%s".',
-                self::LOCALIZABLE,
-                self::NON_LOCALIZABLE,
-                self::BOTH
-            ));
-        }
+        ArgumentChecker::check($referenceType, 'reference-type', [self::LOCALIZABLE, self::NON_LOCALIZABLE, self::BOTH]);
```
Comment (SamirBoulil) [inline src/Command/CreateFamilyCommand.php]: 👍

File: src/Command/MergeAssetsAndVariationsFilesCommand.php [hunk 5]
```diff
@@ -201,6 +220,15 @@ private function isHeaderValid(CsvReader $reader, $row)
     private function getAssetManagerFileHeaders(): array
     {
         $assetHeaders = $this->assetsReader->getHeaders();
+
+        if ($this->withCategories === self::NO) {
+            // Removes the "categories" field from headers
+            $index = array_search(self::CATEGORIES, $assetHeaders);
```
Comment (SamirBoulil) [inline src/Command/MergeAssetsAndVariationsFilesCommand.php:226]: I think there is a one-liner, immutable alternative with array_filter. But this works as well.

File: src/Command/MigrateCommand.php [hunk 6]
```diff
@@ -79,6 +85,21 @@ protected function configure()
                 ),
                 self::AUTO
             )
+            ->addOption('with-categories', null, InputOption::VALUE_OPTIONAL,
+                sprintf(
+                    'Import the categories from your assets file.
+When set to %s, your new asset family will have a categories field, and every asset will contains its former categories.
```
Comment (juliensnz) [inline src/Command/MigrateCommand.php]:
```suggestion
When set to "%s", your new asset family will have a categories field, and every asset will contains its former categories.
```

File: src/Command/MigrateCommand.php [hunk 7]
```diff
@@ -79,6 +85,21 @@ protected function configure()
...
+When set to %s, it will guess the value from the asset file content.
```
Comment (juliensnz) [inline src/Command/MigrateCommand.php]:
```suggestion
When set to "%s", it will guess the value from the asset file content.
```

**Review by juliensnz — APPROVED** [review]

**Review by SamirBoulil — APPROVED** [review]

**Pull request merged**